File: Spring/src/main/java/com/test/mapper/ExtractMapper.java
package com.test.mapper;

import java.text.NumberFormat;
import java.util.ArrayList;
import java.util.List;
import java.util.Locale;

import com.test.model.ExtractModel.Extract;
import com.test.model.ExtractModel.ListaControleLancamento;
import com.test.viewModel.ExtractViewModel;

import org.springframework.stereotype.Service;

@Service
public class ExtractMapper {

    public List<ExtractViewModel> ExtractViewModels(Extract models) {
        List<ExtractViewModel> views = new ArrayList<ExtractViewModel>();

        for (ListaControleLancamento model : models.ListaControleLancamento) {
            ExtractViewModel view = new ExtractViewModel();

            view.DateLaunch = model.DataLancamentoContaCorrenteCliente;
            view.Description = model.LancamentoContaCorrenteCliente.NomeTipoOperacao;
            view.Number = model.NumeroEvento;
            view.Location = model.LancamentoContaCorrenteCliente.NomeSituacaoRemessa;
            view.DateConfirmation = model.DataEfetivaLancamento;
            view.BankName = model.NomeBanco.trim();
            view.BankAgency = model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroAgencia;
            view.BankAccount = model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroContaCorrente;
            view.FinalValue = model.ValorLancamentoRemessa;
            view.FinalValueLocal = NumberFormat.getCurrencyInstance(new Locale("pt", "BR"))
                    .format(model.ValorLancamentoRemessa);
            view.BankDetailsComplete = model.NomeBanco.trim() + "Ag "
                    + model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroAgencia + "CC "
                    + model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroContaCorrente;

            views.add(view);
        }

        return views;
    }
}

File: AspNetCore/TestResource.Domain/Mapper/ExtractMapper.cs
using System.Collections.Generic;
using System.Globalization;
using System.Linq;
using TestResource.Domain.Model;
using TestResource.Domain.ViewModel;

namespace TestResource.Domain.Mapper
{
    public class ExtractMapper
    {
        public List<ExtractViewModel> ExtractViewModels(Extract models)
        {
            return models?.ListaControleLancamento?.Select(model => new ExtractViewModel
            {
                DateLaunch = model.DataLancamentoContaCorrenteCliente,
                Description = model.LancamentoContaCorrenteCliente.NomeTipoOperacao,
                Number = model.NumeroEvento,
                Location = model.LancamentoContaCorrenteCliente.NomeSituacaoRemessa,
                DateConfirmation = model.DataEfetivaLancamento,
                BankName = model.NomeBanco.Trim(),
                BankAgency = model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroAgencia,
                BankAccount = model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroContaCorrente,
                BankDetailsComplete = $"{model.NomeBanco.Trim()} Ag {model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroAgencia} CC {model.LancamentoContaCorrenteCliente.DadosDomicilioBancario.NumeroContaCorrente}",
                FinalValue = model.ValorLancamentoRemessa,
                FinalValueLocal = model.ValorLancamentoRemessa.ToString("C", CultureInfo.CreateSpecificCulture("pt-BR"))
            })
            .ToList();
        }
    }
}
